build/build.sh

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env bash  
+
+# === Script setup
+
+# Change to the directory where the script is located
+cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+
+if [ "$#" -lt 1 ]; then
+	echo "Usage: $0 <directory> <--dry-run> [docker buildx build arguments...]"
+	exit 1
+fi
+
+# Change to the directory for the specified project
+cd "$1" || { echo "Failed to change directory to $1"; exit 1; }
+
+# Remove the first argument, leaving the rest for docker buildx build
+shift
+
+DRY_RUN=false
+PUBLISH=$([[ "$NX_TASK_TARGET_TARGET" == "nx-release-publish" ]] && echo true || echo false)
+BUILD_ARGS=()
+BUILD_ENV_FILE="build.env"
+DISALLOWED_ARGS=(
+	"--firstRelease=true"
+)
+
+for arg in "$@"; do
+	if [[ $arg == "--dry-run" || $arg == "--dryRun=true" ]]; then
+		DRY_RUN=true
+	elif [[ " ${DISALLOWED_ARGS[*]} " =~ $arg ]]; then
+		continue
+	else
+		BUILD_ARGS+=("$arg")
+	fi
+done
+
+# === Setup tags
+
+# publish to Dockerhub
+if [ "$PUBLISH" == true ]; then
+	IMAGE_VERSION=$(jq -r '.version' package.json)
+	IMAGE_VERSION_MAJOR=$(echo "$IMAGE_VERSION" | cut -d '.' -f 1)
+	IMAGE_VERSION_MINOR=$(echo "$IMAGE_VERSION" | cut -d '.' -f 1,2)
+	TAGS+=(
+		"${IMAGE_VERSION}"
+		"${IMAGE_VERSION_MAJOR}"
+		"${IMAGE_VERSION_MINOR}"
+	)
+# build/test in CI
+elif [ "$GITHUB_ACTIONS" == true ]; then
+	TAGS+=(
+		"dev-${GITHUB_RUN_ID}"
+	)
+# local build
+else
+	# When not tagging anything but the image name the "latest" tag is by default applied,
+	# making that explicit here:
+	TAGS+=(
+		"latest"
+	)
+fi
```

Comment:
I think I'd prefer to have explicit flags for that. Would make more clear what is happening from the outside.

Reply:
When docker builds and "loads" it will assume the latest tag if no other tags are appled. We can leave this case out entirely and still the right hings happens. I preferred it explict.